### PR TITLE
Add highlight.js support for dart language

### PR DESCRIPTION
**Pull request by mongodben**
## ✍️ Proposed changes

Add support for Dart language

🎟 _Jira ticket:_ [DOP-2681](https://jira.mongodb.org/browse/DOP-2681)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

## 🧪 How to test changes

unsure (see further comments)

## 💬 Further comments

i didn't figure out how to manually validate that this works. but based on following the pattern of the other supported languages and the [Highlight.js docs](https://highlightjs.org/static/demo/#:~:text=html%3E%3E%0A%0Aquit%20%24%24%24OK-,Dart,-library%20app%3B%0Aimport), i believe that this should work. happy to do some more robust testing here if further guidance is provided on how to do it.

### CODE REVIEW

**Review by DesignerDave — CHANGES_REQUESTED** [review]
Left one small note, but once that's resolved, this PR lgtm @mongodben!

File: .changeset/silent-taxis-peel.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/code': minor
```
Comment (mongodben) [inline .changeset/silent-taxis-peel.md:2]: @DesignerDave this work? (i'm not familiar w the leafygreen release system)

Comment (DesignerDave) [reply .changeset/silent-taxis-peel.md:2]: Yep, perfect!

**Review by DesignerDave — APPROVED** [review]
LGTM! Looks like the linter is holding up the CI checks, so you may want to run `yarn fix` and push another commit to address that so it can be merged.

**Pull request merged**